File: programowanie i podstawy algorytmiki/I zajecia/cpp/haxor.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    string text;
    cin >> text;

    // Metoda 1
    for (char i : text) {
        if (i == 'a') cout << '4';
        else if (i == 'e') cout << '3';
        else if (i == 'i') cout << '1';
        else if (i == 'o') cout << '0';
        else if (i == 's') cout << '5';
        else cout << i;
    }
    cout << "\n";

    // Metoda 2
    vector <vector <char>> arr = {{'a', '4'}};
    for (vector <char> i : arr) {
        replace(text.begin(), text.end(), i[0], i[1]);
    }
    cout << text << "\n";
    return 0;
}

File: programowanie i podstawy algorytmiki/I zajecia/python/haxor.py
text = input()

# # Pierwsza metoda
# result = ""
# for char in text:
#     if char == 'a':
#         result += str(4)
#     elif char == 'e':
#         result += str(3)
#     elif char == 'i':
#         result += str(1)
#     elif char == 'o':
#         result += str(0)
#     elif char == 's':
#         result += str(5)
#     else:
#         result += char
#
# print(result)

# Druga metoda
# code = (('a', 4), ('e', 3), ('i', 1), ('o', 0), ('s', 5))
# for tuple in code:
#     text = text.replace(tuple[0], str(tuple[1]))
# print(text)

# Trzecia metoda
dictionary ={
    'a': 4,
    'e': 3,
    'i': 1,
    'o': 0,
    's': 5
}
# for key in dictionary:
#     text = text.replace(key, str(dictionary[key]))
# print(text)

for key, value in dictionary.items():
    text = text.replace(key, str(value))
print(text)
